spring-boot-3-integrationtests/src/test/java/org/axonframework/spring/utils/MysqlTestContainerExtension.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2010-2023. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.spring.utils;
+
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.springframework.jdbc.datasource.DriverManagerDataSource;
+import org.testcontainers.containers.MySQLContainer;
+
+import javax.sql.DataSource;
+
+/**
+ * Junit 5 extension which fires up a Mysql test container before the test class. After all tests have run, it will tear
+ * it down.
+ * <p>
+ * If you want behavior that creates a container before every individual test, this extension does not support that.
+ */
+public class MysqlTestContainerExtension extends MySQLContainer<MysqlTestContainerExtension>
```

Comment:
Perhaps we can pull this in as a test dependency too i.o. duplicating it?

Reply:
Yes, done.